src/Shared/EnvironmentUtilities.cs

```diff
@@ -12,5 +12,12 @@ internal static partial class EnvironmentUtilities
 
         public static bool Is64BitOperatingSystem =>
             Environment.Is64BitOperatingSystem;
+
+        public static bool IsReservedProperty(string propertyName)
```

Comment:
I'm paralleling our many uses of "reserved" in other places in MSBuild like:
https://github.com/dotnet/msbuild/blob/main/src/Build/Resources/Constants.cs#L102
https://github.com/dotnet/msbuild/blob/main/src/Build/Definition/ProjectProperty.cs#L146
https://github.com/dotnet/msbuild/blob/main/src/Build/ObjectModelRemoting/LinkedObjectFactory.cs#L305

Those are "reserved" in that they are normal properties except that Microsoft specifically recognizes them as special, just as keywords are normal tokens treated as special in programming languages. MSBUILD*, COMPLUS_*, and DOTNET_* are being treated as special, so it makes sense to me that they are also "reserved."

ShouldAlwaysBeLogged makes sense in the context of this PR but does not make sense if you don't know what they are or why they should be logged—what is being explicitly excluded from this?

Reply:
Reserved properties are not normal properties. If you try to set them, you get an error like

```
error MSB4004: The "MSBuildProjectDirectory" property is reserved, and cannot be modified.
```

Since this isn't like that, I don't think it should use the same name.

Reply:
Properties prefixed with an underscore are private but not really. If you try to use one of these as a normal variable, you may get unexpected behavior at some point.

Compromise on IsWellKnownEnvironmentDerivedProperty?

Reply:
`IsWellKnownEnvironmentDerivedProperty` sounds fine to me.